### PR TITLE
Remove ini file parsing

### PR DESCRIPTION
**Pull request by karenzshea**
Part of https://github.com/Project-OSRM/osrm-backend/issues/1971, technically par of the task `read_file_lower_content needs to be renamed` 

:wave: :eyes: @daniel-j-h @TheMarex

### CODE REVIEW

**Comment by daniel-j-h** [comment]
`clang-format` all the files you touched!

File: src/tools/store.cpp [hunk 1]
```diff
@@ -18,10 +17,7 @@ bool generateDataStoreOptions(const int argc,
     // declare a group of options that will be allowed only on command line
     boost::program_options::options_description generic_options("Options");
     generic_options.add_options()("version,v", "Show version")("help,h", "Show this help message")(
-        "springclean,s", "Remove all regions in shared memory")(
-        "config,c", boost::program_options::value<boost::filesystem::path>(&paths["config"])
-                        ->default_value("server.ini"),
-        "Path to a configuration file");
+        "springclean,s", "Remove all regions in shared memory");
```
Comment (daniel-j-h) [inline src/tools/store.cpp]: If you `clang-format` and want to force the options to appear on a new line, you can do it like this:

    add_option()                     //
      ("version", "Show version")    //
      ("help", "Show help");         //

File: features/options/prepare/help.feature [hunk 2]
```diff
@@ -11,14 +11,13 @@ Feature: osrm-prepare command line options: help
...
         And stdout should contain "--help"
-        And stdout should contain "--config"
         And stdout should contain "Configuration:"
         And stdout should contain "--profile"
         And stdout should contain "--threads"
         And stdout should contain "--core"
         And stdout should contain "--level-cache"
         And stdout should contain "--segment-speed-file"
-        And stdout should contain 21 lines
+        And stdout should contain 18 lines
```
Comment (MoKob) [inline features/options/prepare/help.feature]: On my system osrm-prepare only has 2 lines for --config, where does the third line come from?

Comment (MoKob) [inline features/options/prepare/help.feature]: never mind, found it below

**Comment by TheMarex** [comment]
:+1: looks good to merge.

**Pull request merged**